src/classes/Hotel.js

```diff
@@ -1,44 +1,91 @@
+import Booking from "./Booking.js";
+import Room from "./Room.js";
+import Customer from "./Customer.js";
+var dayjs = require('dayjs')
+let moment = require('moment')
 
 class Hotel {
-  constructor(bookingData, roomData, customerData, currentCustomer) {
-    this.currentDate = ''
-    this.bookingData = bookingData
-    this.roomData = roomData
-    this.customerData = customerData
+  constructor(bookingData = [], roomData = [], customerData = [], id) {
+    this.currentDate = moment().format('YYYY/MM/DD');
     this.allBookings = []
+    this.allRooms = []
+    this.allCustomers = []
+    this.currentCustomerID = id
+    this.currentCustomer = {}
+    this.getCurrentDate()
+    this.updateAllRooms(roomData)
+    this.updateAllBookings(bookingData)
+    this.addRoom()
+    this.updateAllCustomers(customerData)
+    this.addCurrentCustomer(id)
+    this.addBookings()
+  }
 
-    this.currentCustomerId = currentCustomer.id
-    this.currentCustomer = currentCustomer
-    this.currentCustomerBookings = null
+  updateAllRooms(somedata) {
```

Comment:
On these methods where you push in new classes, you may want to clear out the array before you do to avoid duplicates in the future if you ever call this method a second time